apm-sniffer/apm-sdk-plugin/tomcat-7.x-8.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/tomcat78x/TomcatInvokeInterceptor.java

```diff
@@ -83,32 +83,26 @@ public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allAr
         SpanLayer.asHttp(span);
 
         if (Config.Plugin.Tomcat.COLLECT_HTTP_PARAMS) {
-            final Map<String, String[]> parameterMap = new HashMap<>();
-            final org.apache.coyote.Request coyoteRequest = request.getCoyoteRequest();
-            final Parameters parameters = coyoteRequest.getParameters();
-            for (final Enumeration<String> names = parameters.getParameterNames(); names.hasMoreElements(); ) {
-                final String name = names.nextElement();
-                parameterMap.put(name, parameters.getParameterValues(name));
-            }
-
-            if (!parameterMap.isEmpty()) {
-                String tagValue = CollectionUtil.toString(parameterMap);
-                tagValue = Config.Plugin.Http.HTTP_PARAMS_LENGTH_THRESHOLD > 0 ? StringUtil.cut(tagValue, Config.Plugin.Http.HTTP_PARAMS_LENGTH_THRESHOLD) : tagValue;
-                Tags.HTTP.PARAMS.set(span, tagValue);
-            }
+            collectHttpParam(request, span);
         }
     }
 
     @Override
     public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
         Object ret) throws Throwable {
+        Request request = (Request) allArguments[0];
         HttpServletResponse response = (HttpServletResponse) allArguments[1];
 
         AbstractSpan span = ContextManager.activeSpan();
         if (IS_SERVLET_GET_STATUS_METHOD_EXIST && response.getStatus() >= 400) {
             span.errorOccurred();
             Tags.STATUS_CODE.set(span, Integer.toString(response.getStatus()));
         }
+        // Active HTTP parameter collection automatically in the profiling context.
+        // https://github.com/apache/skywalking/issues/4542
```

Comment:
Don't refer to issue, unless there is something special, because if don't follow this, we will have thousands of issue links in the codes.